pytorch_lightning/metrics/metric.py

```diff
@@ -40,32 +44,41 @@ class Metric(DeviceDtypeModuleMixin, nn.Module, ABC):
 
         * input_convert: pre-forward hook that takes care of input conversion
         * output_convert: post-forward hook that takes care of output convertion
-        * ddp_sync: implementation of ddp sync, default is gather all
-        * aggregate: implement how values should be aggregated
+        * ddp_reduce: implementation of ddp sync + aggregation, default is ddp_sync + aggregate
         * compute: post-ddp sync for additional metric computations
 
+    ``ddp_reduce`` by default calls the following methods, which can also be overwritten if necessary.
+
+        * ddp_sync: implements how values should be synced across ddp-processes. Defaults to gather all.
+        * aggregate: implement how values should be aggregated (defaults to mean).
+
     Call order
 
-        input_convert -> forward -> output_convert -> ddp_sync -> aggregate -> compute
+        input_convert -> forward -> output_convert -> ddp_reduce (per default being ddp_sync -> aggregate) -> compute
 
     """
 
-    def __init__(self, name: str):
+    def __init__(self, name: str, reduce_group: Optional[Any] = None):
         """
         Args:
             name: the metric's name
+            reduce_group: the process group for DDP reduces (only needed for DDP training).
+                Defaults to all processes (world)
 
         """
         super().__init__()
         self.name = name
         self._dtype = torch.get_default_dtype()
-        self._device = torch.device('cpu')
+        self._device = torch.device("cpu")
+
+        self.reduce_group = reduce_group
+
+        self._step_vals = []
```

Comment:
Is it possible for us to make this a buffer, so saving the model and thereby the metric will also save the current internal state of the metric

Reply:
Actually this will be difficult, since buffers can only be tensors and tensors have a fixed size. Also buffers are moved to the device accordingly. We probably don't want this, but want it to be on cpu for the time of storing and want to move it back to the device after aggregation.

On the other hand, I'm not sure if it won't be saved anyway, since pytorch uses basically pickle for saving.

Reply:
Alright, good point.
Even IF they are not saved I don't see it as a huge drawback, more in sense that it would be nice to have